### PR TITLE
Feature/e2e rust get nonce

### PR DESCRIPTION
**Pull request by tchataigner**
Add test cases in rust for getNonce route.

Parent issue: #904 

# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Testing

## What is the current behavior?

Resolves: #NA

## What is the new behavior?

In this draft state it contains working behavior for:
- [x] Return an error for a nonexisting block
- [x] Return a nonce 0 for a non-used contract address
- [x] Return a nonce 0 if the contract deployed at this address is not an account contract
- [x] Otherwise, return the nonce of the account deployed at this address

## Does this introduce a breaking change?

No

## Other information

The tested behavior differs from #904 checklist but after some discussions the one implemented is the correct one.

### CODE REVIEW

**Comment by tchataigner** [comment]
Hey @tdelabro, I am currently facing some behavior that goes against what was mentioned in #904. 

Currently when I test for errors on non-used contract address or non-account contract I do not get errors but an `Ok()` with a nonce 0.

Which one is the correct one? The one from the issue or the one I'm experiencing?

You should be able to test that by running the `fail_non_used_contract_address` and `fail_non_account_contract` tests.

**Comment by tdelabro** [comment]
@tchataigner Hey,
I think this has been done in order to respect Starknet current behavior. Any non-initialized memory slot of a contract will return 0 if read.
So nothing to fix, the behaviour I described in the issue was incorrect. 
You can put the PR as ready it that was the only thing bothering you :)

**Comment by tchataigner** [comment]
Alright ! Thanks, will clean up the code, fix tests and I'll set it as ready.

File: starknet-rpc-test/get_nonce.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+    assert_eq!(
+        rpc.get_nonce(
+            BlockId::Number(0),
+            FieldElement::from_hex_be("0x4269DEADBEEF").expect("Invalid Contract Address")
+        )
+        .await
+        .ok(),
+        Some(
+            FieldElement::from_hex_be("0x0000000000000000000000000000000000000000000000000000000000000000")
```
Comment (tdelabro) [inline starknet-rpc-test/get_nonce.rs:58]: `FieldElement::ZERO` instead

File: starknet-rpc-test/get_nonce.rs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            argent_account_address,
+            FieldElement::from_hex_be(MINT_AMOUNT).expect("Invalid Mint Amount"),
+            None,
+        )])
+        .await?;
+
+    assert_eq!(
+        rpc.get_nonce(BlockId::Tag(BlockTag::Latest), account.address(),).await.ok(),
+        Some(
+            FieldElement::from_hex_be("0x0000000000000000000000000000000000000000000000000000000000000001")
```
Comment (tdelabro) [inline starknet-rpc-test/get_nonce.rs:109]: `FieldElement::ONE`

**Review by tdelabro — CHANGES_REQUESTED** [review]

**Review by tdelabro — APPROVED** [review]

**Pull request merged**